docs/content/concepts/resources.mdx

```diff
@@ -5,357 +5,355 @@ description: Resources enable you to separate graph logic from environment, and
+defs = Definitions(
+    jobs=[update_service_job],
+    resources={
+        "my_conn": MyConnectionResource(username="my_user"),
+    },
+)
 ```
 
-#### Providing resources to ops
+</TabItem>
+</TabGroup>
 
-Jobs provide resources to the ops inside them. A job has a dictionary that maps resource keys to resource definitions. You can supply this dictionary to the `resource_defs` argument when using either of the ways to construct a job: <PyObject object="GraphDefinition" method="to_job" /> or <PyObject object="job" decorator />.
+There are many supported config types that can be used when defining resources. See the [advanced config types documentation](/concepts/configuration/advanced-config-types) for a more comprehensive overview on the available config types.
 
-Supplying resources when using <PyObject object="GraphDefinition" method="to_job" /> is especially common, because you can build multiple jobs from the same graph that are distinguished by their different resources.
+### Using environment variables with resources
 
-```python file=/concepts/resources/resources.py startafter=start_graph_example endbefore=end_graph_example
-from dagster import graph
+Resources can be configured using environment variables, which is useful for secrets or other environment-specific configuration. If you're using Dagster Cloud, environment variables can be [configured directly in the UI](/dagster-cloud/managing-deployments/environment-variables-and-secrets).
 
+To use environment variables, pass an <PyObject object="EnvVar" /> when constructing your resource.
 
-@graph
-def do_database_stuff():
-    op_requires_resources()
+```python file=/concepts/resources/pythonic_resources.py startafter=start_new_resources_env_vars endbefore=end_new_resources_env_vars dedent=4
+from dagster import EnvVar, Definitions, ConfigurableResource
 
+class CredentialsResource(ConfigurableResource):
+    username: str
+    password: str
 
-do_database_stuff_prod = do_database_stuff.to_job(
-    resource_defs={"database": database_resource_a}
-)
-do_database_stuff_dev = do_database_stuff.to_job(
-    resource_defs={"database": database_resource_b}
+defs = Definitions(
+    assets=...,
+    resources={
+        "credentials": CredentialsResource(
+            username=EnvVar("MY_USERNAME"),
+            password=EnvVar("MY_PASSWORD"),
+        )
+    },
 )
 ```
 
-Supplying resources to the <PyObject object="job" decorator />, i.e. when there aren't multiple jobs for the same graph, is also useful. For example, if you want to use an off-the-shelf resource or supply configuration in one place instead of in every op.
+For more information on using environment variables with Dagster, refer to the [Environment variables guide](/guides/dagster/using-environment-variables-and-secrets).
 
-```python file=/concepts/resources/resources.py startafter=start_job_example endbefore=end_job_example
-from dagster import job
+### Configuring resources at runtime
```

Comment:
IMO we should change our language from "runtime" to "launch time" in order to disambiguate from when the python code is running to create definitions, which is also "runtime" from a different POV.